transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java

```diff
@@ -14,10 +14,41 @@
 
 package com.google.android.datatransport;
 
+import java.util.Arrays;
+
 public enum Priority {
   /** Quality of the service is within an hour. */
-  DEFAULT,
+  DEFAULT(0),
+
+  /** Events delivered at most daily, on unmetered network. */
+  VERY_LOW(1),
+
+  /** Events delivered as soon as possible. */
+  HIGHEST(2),
+  ;
+
+  private static final Priority[] PRIORITY_MAP;
+
+  static {
+    // the order of this array must match the values of the enums.
+    PRIORITY_MAP = Priority.values();
+    Arrays.sort(PRIORITY_MAP, (p1, p2) -> p1.value - p2.value);
+  }
+
+  private final int value;
+
+  Priority(int value) {
+    this.value = value;
+  }
+
+  public int getValue() {
+    return value;
+  }
 
-  /** Events delivered at most daily, on unmetered network */
-  VERY_LOW,
+  public static Priority forValue(int value) {
+    if (value < 0 || value >= PRIORITY_MAP.length) {
```

Comment:
For this to work, values would have to increment monotonically.  Either comment that in the `enum`, or use check the last value of the array as max.

Reply:
I fixed the underlying issue that you pointed out. Note that I also removed the int -> priority logic into transport-runtime.